apps/dashboard/src/components/engine/status/components/ErrorRate.tsx

```diff
@@ -0,0 +1,112 @@
+"use client";
+
+import { useQuery } from "@tanstack/react-query";
+import { Area, AreaChart, CartesianGrid, XAxis, YAxis } from "recharts";
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import {
+  type ChartConfig,
+  ChartContainer,
+  ChartLegend,
+  ChartLegendContent,
+  ChartTooltip,
+  ChartTooltipContent,
+} from "@/components/ui/chart";
+
+export function ErrorRate() {
+  const monitorData = useQuery(["errorRate"], async () => {
+    const res = await fetch("/api/engine-status/error-rate", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({}),
+    });
+
+    const json = await res.json();
+    return json;
+  });
+
+  const chartConfig: { [key: string]: { label: string; color: string } } = {
+    success: {
+      label: "Success",
+      color: "hsl(var(--chart-2))",
+    },
+    failures: {
+      label: "Failures",
+      color: "hsl(var(--chart-5))",
+    },
+  } satisfies ChartConfig;
+
+  const getAreaCharts = () => {
+    const areaCharts: JSX.Element[] = [];
+    if (monitorData.data?.tags === undefined) {
+      return areaCharts;
+    }
+    for (const tag of monitorData.data.tags) {
+      areaCharts.push(
+        <Area
+          key={tag}
+          dataKey={tag}
+          type="natural"
+          fill={`var(--color-${tag})`}
+          fillOpacity={0.3}
+          stroke={`var(--color-${tag})`}
+          stackId="a"
+        />,
+      );
+    }
+    return areaCharts;
+  };
```

Comment:
refactor this into a `useMemo(() => ...)` - and simply render the result of it below

Reply:
or alternatively you can do this in a ternary directly inline

Reply:
done ✅